tools/code_coverage/python_coverage.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python
+
+"""
+usage: python_coverage.py > python-coverage.info
+"""
+
+from os import path
+from xml.etree import ElementTree
+
+
+def process_coverage_file(xml_file):
+    tree = ElementTree.parse(xml_file)
+    root = tree.getroot()
+
+    sources = root.findall("sources/source")
+
+    source = sources[-1].text
+
+    for cls in root.findall("packages/package/classes/class"):
+        cls_filename = cls.attrib.get("filename")
+        cls_filename = path.join(source, cls_filename)
+
+        if not path.exists(cls_filename):
+            continue
+
+        print("TN:")
+        print(f"SF:{cls_filename}")
+
+        branch_index = 0
+        for line in cls.findall("lines/line"):
+            line_hits = line.attrib.get("hits")
+            line_number = line.attrib.get("number")
+            line_branch = line.attrib.get("branch")
```

Comment:
When will a a line have this attribute 'branch'?

Reply:
no branch coverage for now